internal/security/fileprovider/config.go

```diff
@@ -0,0 +1,93 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under
+// the License.
+//
+// SPDX-License-Identifier: Apache-2.0'
+//
+
+package fileprovider
+
+import (
+	"github.com/edgexfoundry/edgex-go/internal/pkg/bootstrap/interfaces"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/config"
+	"github.com/edgexfoundry/edgex-go/internal/security/secretstoreclient"
+)
+
+var Configuration = &ConfigurationStruct{}
+
+type ConfigurationStruct struct {
+	Writable          WritableInfo
+	Logging           config.LoggingInfo
+	SecretService     secretstoreclient.SecretServiceInfo
+	TokenFileProvider TokenFileProviderInfo
+	Clients           map[string]config.ClientInfo
+	Registry          config.RegistryInfo
+	Service           config.ServiceInfo
+}
+
+type WritableInfo struct {
+	LogLevel string
+}
+
+// UpdateFromRaw converts configuration received from the registry to a service-specific configuration struct which is
+// then used to overwrite the service's existing configuration struct.
+func (c *ConfigurationStruct) UpdateFromRaw(rawConfig interface{}) bool {
+	configuration, ok := rawConfig.(*ConfigurationStruct)
+	if ok {
+		// Check that information was successfully read from Registry
+		if configuration.SecretService.Server == "" {
+			return false
+		}
+		c = configuration
```

Comment:
Not sure this is going to have a deep copy of `configuration` into `c`. it might be ineffective and lose the assigned values after this call. I think you should do 
```go
*c = *configuration
```

Reply:
Bug is legit.  This was part of the boilerplate and was moved to config/config.go. This file needs to be deleted.

Reply:
File removed.